File: evidence/generated-systems/angular/manufacturing-on-angular/src/app/models/Employee.js

// Define collection and schema for Employee
export interface Employee {
    firstName:
	type : string
    lastName:
	type : string
    WorkCenter:
	type : Schema.Types.ObjectId
    ShiftAssignments:
 	type : [{ type: Schema.Types.ObjectId, ref: 'ShiftAssignment' }]
    CorrectiveActions:
 	type : [{ type: Schema.Types.ObjectId, ref: 'CorrectiveAction' }]
    Role:
 	type : String
    SkillLevel:
 	type : String
#
    collection: 'employees'
}

File: evidence/generated-systems/rubyonrails/manufacturing-on-rails/manufacturingonrails/app/models/employee.rb
class Employee < ApplicationRecord
  enum Role: [:Operator, :Technician, :Supervisor, :Planner, :QualityEngineer, :Buyer]
  enum SkillLevel: [:Novice, :Competent, :Proficient, :Expert]


  has_many :WorkCenter, class_name: 'WorkCenter'
  has_many :ShiftAssignments, class_name: 'ShiftAssignment'
  has_many :CorrectiveActions, class_name: 'CorrectiveAction'

end
